libbpf-cargo/src/btf/btf.rs

```diff
@@ -17,6 +18,8 @@ pub struct Btf<'a> {
     ptr_size: u32,
     string_table: &'a [u8],
     bpf_obj: *mut libbpf_sys::bpf_object,
+    anon_union_count: Mutex<u32>,
+    anon_struct_count: Mutex<u32>,
```

Comment:
I'm assuming you're using a `Mutex` to avoid the borrowing issues and to not change method signatures from `&self` -> `&mut self`. Mutex is kinda weird semantically. `AtomicU32` is better but it's still weird. I think we should just change the signatures to `&mut self`, use a regular `u32`, and skip the helper (it doesn't do much anyways). See below for my suggestion

Reply:
Yes, I was hesitant to change the method signatures.  Fixed.